packages/flutter/lib/src/material/icon.dart

```diff
@@ -22,24 +22,26 @@ const Map<IconSize, int> _kIconSize = const <IconSize, int>{
   IconSize.s48: 48,
 };
 
-class Icon extends StatelessComponent {
-  Icon({
-    Key key,
-    this.size: IconSize.s24,
-    this.icon: '',
-    this.color,
-    this.colorFilter
-  }) : super(key: key) {
-    assert(size != null);
-    assert(icon != null);
+class _MaterialDensityQualifier extends BreakpointQualifier {
+  _MaterialDensityQualifier(BuildContext context) : super(
+    candidates: {
+      1.0 : 'mdpi',
+      1.5 : 'hdpi',
+      2.0 : 'xhdpi',
+      3.0 : 'xxhdpi',
+      4.0 : 'xxxhdpi'
+    }
+  ) {
+    MediaQueryData media = MediaQuery.of(context);
```

Comment:
I'm very concerned about calling `.of()` functions for `Inherited`s outside of the actual class that ends up being marked as a dependency. Even if we get it right, it's going to be copy-pasta'ed into buggy code, because it hides the implications. I think you should do the getting of the media in the build function and pass it to the qualifier.